File: merchant-integration-api/src/foam.py
import foam
from config import FOAM_CONFIG

foam.init(
    service_name="merchant-integration-api",
    api_key=FOAM_CONFIG['api_key'],
    is_production=FOAM_CONFIG['is_production'],
    sample_rate=FOAM_CONFIG.get('sample_rate', 1.0),
)

File: vcn-gateway/src/foam.kt
package com.affirm.vcn

import ai.foam.opentelemetry.FoamSdk
import ai.foam.opentelemetry.FoamConfig
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import javax.annotation.PostConstruct

@Configuration
class FoamConfiguration {
    @Value("\${foam.api.key}")
    private lateinit var apiKey: String

    @Value("\${spring.profiles.active}")
    private lateinit var activeProfile: String

    @PostConstruct
    fun initFoam() {
        FoamSdk.init(
            FoamConfig(
                serviceName = "vcn-gateway",
                apiKey = apiKey,
                isProduction = activeProfile == "production"
            )
        )
    }
}